### PR TITLE
tracing macro overhaul

### PR DESCRIPTION
**Pull request by nashif**
See #33603 for more details.

Most of the commits are from #33603 with additional adaption and enhancements for ctf and systemview

﻿- Tracing: Trace hook support macros
- Tracing: Semaphore tracing
- Tracing: Semaphore tracing documentation
- Tracing: Mutex tracing hooks
- Tracing: Conditional variable tracing
- Tracing: Thread tracing
- Tracing: Queue tracing
- Tracing: FIFO Tracing
- Tracing: LIFO Tracing
- Tracing: Memory Stack tracing
- Tracing: Message Queue tracing
- Tracing: Mailbox tracing
- Tracing: Pipe tracing
- Tracing: Memory Heap tracing
- Tracing: Memory Slab tracing
- Tracing: Timer tracing
- Tracing: Poll API and Work Poll tracing
- Tracing: Work Queue tracing
- tracing: add empty test macros
- tracing: Use new macros with CTF based tracing
- samples: philosopher: add thread names
- tracing: support new macros with systemview

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
No real objections, but why the `_PORT_` in all the macros?

File: include/kernel.h [hunk 1]
```diff
@@ -1934,10 +1919,11 @@ static inline int z_impl_k_queue_is_empty(struct k_queue *queue)
  */
 __syscall void *k_queue_peek_head(struct k_queue *queue);
 
-static inline void *z_impl_k_queue_peek_head(struct k_queue *queue)
+/*static inline void *z_impl_k_queue_peek_head(struct k_queue *queue)
```
Comment (tejlmand) [inline include/kernel.h]: This code commented out here, just to be removed later in this commit: https://github.com/zephyrproject-rtos/zephyr/pull/34843/commits/e9c9e73e5d62134037831dce3c960326d516e3b9

Why not remove it directly in https://github.com/zephyrproject-rtos/zephyr/pull/34843/commits/ef9c5f0f5b1b1bfe959b00c4d376fd915a17c898.

**Review by tejlmand — CHANGES_REQUESTED** [review]
A couple of small observations, and then a request to make PERCEPIO_TRACERECORDER depending on the module being present.

File: subsys/tracing/Kconfig [hunk 2]
```diff
@@ -32,6 +32,13 @@ config TRACING_NONE
 	help
 	  None of the available tracing formats is selected.
 
+config PERCEPIO_TRACERECORDER
```
[the remark this answers is not in the document]
Comment (tejlmand) [reply subsys/tracing/Kconfig:35]: > Adding such dependency requires a corresponding file `modules/TraceRecorder/Kconfig` with content:
> 
> ```
> config ZEPHYR_TRACERECORDER_MODULE
> 	bool
> ```

@nashif you missed this part, so I created a followup PR here: #35032

**Pull request merged**